src/client.c

```diff
@@ -1,46 +1,175 @@
+    pthread_mutex_unlock(&global_mutex);
+
+    client->res.success = reserved;
     return NULL;
 }
 
-client_t* client_new(movie_t* movie, int id) {
+// Bloqueo por Fila
+void* client_per_row_mutex(void* arg) {
+    client_t* client = (client_t*)arg;
+    movie_t* movie   = client->data.movie;
+    int id           = client->data.id;
+    bool reserved    = false;
+    int row          = id % movie->nrows;
+
+    pthread_mutex_lock(&row_mutexes[row]); // Bloqueo de fila
+    int rand_col = rand() % movie->ncols;
+    reserved     = movie_reserve_seat(movie, rand_col, row, id);
+    pthread_mutex_unlock(&row_mutexes[row]);
+
+    client->res.success = reserved;
+    return NULL;
+}
+
+// Bloqueo por Asiento
+void* client_per_seat_mutex(void* arg) {
+    client_t* client = (client_t*)arg;
+    movie_t* movie   = client->data.movie;
+    int id           = client->data.id;
+    bool reserved    = false;
+
+    int rand_index = rand() % (movie->nrows * movie->ncols);
+    int rand_row   = rand_index / movie->ncols;
+    int rand_col   = rand_index % movie->ncols;
+
+    pthread_mutex_lock(&seat_mutexes[rand_index]); // Bloqueo de asiento
+    reserved = movie_reserve_seat(movie, rand_col, rand_row, id);
+    pthread_mutex_unlock(&seat_mutexes[rand_index]);
+
+    client->res.success = reserved;
+    return NULL;
+}
+
+client_t* client_new(movie_t* movie, int id, int method) {
     client_t* client = (client_t*)malloc(sizeof(client_t));
     if (client == NULL) {
         return NULL;
     }
 
+    switch (method) {
+    case 0:
+        client_run = client_global_mutex;
+        break;
+    case 1:
+        client_run = client_per_row_mutex;
+        break;
+    case 2:
+        client_run = client_per_seat_mutex;
+        break;
+    default:
+        client_run = NULL;
+        free(client);
+        return NULL;
+    }
+
     client_data_init(&client->data, movie, id);
     client_res_init(&client->res);
 
     return client;
 }
 
+// Inicia el hilo del cliente.
 int client_start(client_t* client) {
     assert(client != NULL);
-    return pthread_create(&client->thread, NULL, (void* (*)(void*))client_run, &client->data);
+    return pthread_create(&client->thread, NULL, client_run, client);
 }
 
+// Libera la memoria ocupada por el cliente.
 void client_free(client_t* client) {
-    free(client);
+    if (client != NULL) {
+        free(client);
+    }
 }
 
+// Inicializa los datos del cliente.
 void client_init(client_t* client, movie_t* movie, int id) {
     assert(client != NULL);
     assert(movie != NULL);
 
     client_data_init(&client->data, movie, id);
     client_res_init(&client->res);
 }
+
+// Inicializa los mutexes según el método de reserva seleccionado.
+void client_init_mutexes(int method, int nrows, int ncols) {
+    switch (method) {
+    case 0:
+        pthread_mutex_init(&global_mutex, NULL);
+        break;
+    case 1:
+        row_mutexes = malloc(nrows * sizeof(pthread_mutex_t));
```

Comment:
Siempre que hagas un `malloc` o `calloc`  comprueba que haya sido exitoso por favor, en general nunca falla, pero nunca está demás ser cuidadoso. En este caso también deberíamos agregar un valor de retorno para indicar si el código debe continuar ejecutando o detenerse.
```suggestion
        row_mutexes = malloc(nrows * sizeof(pthread_mutex_t));
        if (row_mutexes == NULL) {
            return false;
        }
```